Remotion/Data/DomainObjects/Persistence/Rdbms/SchemaGeneration/RdbmsStructuredTypeDefinitionProvider.cs

```diff
@@ -0,0 +1,102 @@
+                 CreateTypeDefinition(typeof(Byte?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Byte?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(DateTime?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(DateTime?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Decimal?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Decimal?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Double?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Double?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Guid?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Guid?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Int16?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Int16?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Int32?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Int32?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Int64?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Int64?), true, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Single?), false, storageTypeInformationProvider),
+                 CreateTypeDefinition(typeof(Single?), true, storageTypeInformationProvider)
+             };
+    }
+
+    private IRdbmsStructuredTypeDefinition CreateTypeDefinition (Type dotnetType, bool withUniqueConstraint, IStorageTypeInformationProvider storageTypeInformationProvider)
+    {
+      var storageTypeInformation = storageTypeInformationProvider.GetStorageType(dotnetType);
+      return CreateTypeDefinition(storageTypeInformation, withUniqueConstraint);
+    }
+
+    private IRdbmsStructuredTypeDefinition CreateTypeDefinition (IStorageTypeInformation storageTypeInformation, bool withUniqueConstraint)
+    {
+      var dbType = storageTypeInformation.StorageDbType;
+      var typeNameDefinition = new EntityNameDefinition(null, $"TVP_{dbType}{(withUniqueConstraint ? "_distinct" : null)}");
+      var columnDefinition = new ColumnDefinition("Value", storageTypeInformation, false);
+      var propertyDefinitions = new[] { new SimpleStoragePropertyDefinition(storageTypeInformation.DotNetType, columnDefinition) };
+
+      var tableConstraintDefinitions = Array.Empty<ITableConstraintDefinition>();
+      if (withUniqueConstraint)
+      {
+        tableConstraintDefinitions =
+        [
+            new UniqueConstraintDefinition($"UQ_{typeNameDefinition.EntityName}_{columnDefinition.Name}", true, new[] { columnDefinition })
```

Comment:
"UQ" seems arbitraty. "UC" would feel more like the expected naming.

Reply:
the usual prefixes seem to be UQ or AK (alternate key) -https://stackoverflow.com/questions/4836391/naming-convention-for-unique-constraint
Automatically generated constraints for table-types begin with "UQ_"